docs: scaling task agents (#70)
encord-team/encord-agents: docs/task_agents/queue_runner.md

### Modal Integration

```python title="modal_agent.py"
import modal
from encord_agents.tasks import QueueRunner

stub = modal.Stub("agent-tasks")
runner = QueueRunner(project_hash="<your_project_hash>")

@runner.stage("my_stage")
def process_task(task: AgentTask) -> str:
    # Your processing logic
    return "next_stage"

@stub.function
def modal_task(task_spec: str) -> str:
    return process_task(task_spec)

@stub.local_entrypoint()
def main():
    for stage in runner.get_agent_stages():
        for task in stage.get_tasks():
            modal_task.remote(task.model_dump_json())
```

!!! tip
    For a more detailed example, see the [Modal example](https://github.com/encord-team/encord-agents/blob/main/docs/code_examples/modal/queue_runner_example.py){target="\_blank" rel="noopener noreferrer"}.

## Key Concepts

### Task Specification Format

The `QueueRunner` uses JSON strings to pass task specifications between processes:

```python
# Converting task to JSON spec
task_spec = task.model_dump_json()

# The wrapped agent function handles parsing and dependency injection
result = my_agent(task_spec)  # Returns TaskCompletionResult as JSON
```

> A task specification will be a string that looks like this:

Reviewer: ```suggestion
A task specification is a string in the following format:
```
